--- M2Crypto-0.30.1/SWIG/_m2crypto.i~	2018-02-24 13:54:59.000000000 +0000
+++ M2Crypto-0.30.1/SWIG/_m2crypto.i	2018-09-14 14:37:34.098789876 +0000
@@ -78,7 +78,6 @@
 %include _rand.i
 %include _evp.i
 %include _aes.i
-%include _rc4.i
 %include _dh.i
 %include _rsa.i
 %include _dsa.i
--- M2Crypto-0.30.1/M2Crypto.egg-info/SOURCES.txt~	2018-04-29 18:47:37.000000000 +0000
+++ M2Crypto-0.30.1/M2Crypto.egg-info/SOURCES.txt	2018-09-14 14:37:51.994789627 +0000
@@ -68,7 +68,6 @@
 SWIG/_pkcs7.i
 SWIG/_py3k_compat.i
 SWIG/_rand.i
-SWIG/_rc4.i
 SWIG/_rsa.i
 SWIG/_ssl.i
 SWIG/_threads.i
--- M2Crypto-0.30.1/SWIG/_m2crypto_wrap.c~	2018-04-29 18:47:31.000000000 +0000
+++ M2Crypto-0.30.1/SWIG/_m2crypto_wrap.c	2018-09-14 14:40:20.901787554 +0000
@@ -6444,56 +6444,6 @@
 }
 
 
-#include <openssl/rc4.h>
-
-
-RC4_KEY *rc4_new(void) {
-    RC4_KEY *key;
-
-    if (!(key = (RC4_KEY *)PyMem_Malloc(sizeof(RC4_KEY))))
-        PyErr_SetString(PyExc_MemoryError, "rc4_new");
-    return key;
-}
-
-void rc4_free(RC4_KEY *key) {
-    PyMem_Free((void *)key);
-}
-
-PyObject *rc4_set_key(RC4_KEY *key, PyObject *value) {
-    const void *vbuf;
-    int vlen = 0;
-
-    if (m2_PyObject_AsReadBufferInt(value, &vbuf, &vlen) == -1)
-        return NULL;
-
-    RC4_set_key(key, vlen, vbuf);
-    Py_RETURN_NONE;
-}
-
-PyObject *rc4_update(RC4_KEY *key, PyObject *in) {
-    PyObject *ret;
-    const void *buf;
-    Py_ssize_t len;
-    void *out;
-
-    if (PyObject_AsReadBuffer(in, &buf, &len) == -1)
-        return NULL;
-
-    if (!(out = PyMem_Malloc(len))) {
-        PyErr_SetString(PyExc_MemoryError, "expected a string object");
-        return NULL;
-    }
-    RC4(key, len, buf, out);
-
-    ret = PyBytes_FromStringAndSize(out, len);
-
-    PyMem_Free(out);
-    return ret;
-}
-
-int rc4_type_check(RC4_KEY *key) {
-    return 1;
-}
 
 
 #include <openssl/bn.h>
--- M2Crypto-0.30.1/SWIG/_m2crypto_wrap.c~	2018-09-14 14:40:20.000000000 +0000
+++ M2Crypto-0.30.1/SWIG/_m2crypto_wrap.c	2018-09-14 14:43:07.230785238 +0000
@@ -13202,17 +13202,6 @@
 }
 
 
-SWIGINTERN PyObject *_wrap_rc4(PyObject *self, PyObject *args) {
-  PyObject *resultobj = 0;
-  EVP_CIPHER *result = 0 ;
-  
-  result = (EVP_CIPHER *)EVP_rc4();
-  resultobj = SWIG_NewPointerObj(SWIG_as_voidptr(result), SWIGTYPE_p_EVP_CIPHER, 0 |  0 );
-  return resultobj;
-fail:
-  return NULL;
-}
-
 
 SWIGINTERN PyObject *_wrap_rc2_40_cbc(PyObject *self, PyObject *args) {
   PyObject *resultobj = 0;
@@ -15059,141 +15048,6 @@
 }
 
 
-SWIGINTERN PyObject *_wrap_rc4_new(PyObject *self, PyObject *args) {
-  PyObject *resultobj = 0;
-  RC4_KEY *result = 0 ;
-  
-  result = (RC4_KEY *)rc4_new();
-  resultobj = SWIG_NewPointerObj(SWIG_as_voidptr(result), SWIGTYPE_p_RC4_KEY, 0 |  0 );
-  return resultobj;
-fail:
-  return NULL;
-}
-
-
-SWIGINTERN PyObject *_wrap_rc4_free(PyObject *self, PyObject *args) {
-  PyObject *resultobj = 0;
-  RC4_KEY *arg1 = (RC4_KEY *) 0 ;
-  void *argp1 = 0 ;
-  int res1 = 0 ;
-  PyObject * obj0 = 0 ;
-  
-  if(!PyArg_UnpackTuple(args,(char *)"rc4_free",1,1,&obj0)) SWIG_fail;
-  res1 = SWIG_ConvertPtr(obj0, &argp1,SWIGTYPE_p_RC4_KEY, 0 |  0 );
-  if (!SWIG_IsOK(res1)) {
-    SWIG_exception_fail(SWIG_ArgError(res1), "in method '" "rc4_free" "', argument " "1"" of type '" "RC4_KEY *""'"); 
-  }
-  arg1 = (RC4_KEY *)(argp1);
-  {
-    if (!arg1) {
-      SWIG_exception(SWIG_ValueError,"Received a NULL pointer.");
-    }
-  }
-  rc4_free(arg1);
-  resultobj = SWIG_Py_Void();
-  return resultobj;
-fail:
-  return NULL;
-}
-
-
-SWIGINTERN PyObject *_wrap_rc4_set_key(PyObject *self, PyObject *args) {
-  PyObject *resultobj = 0;
-  RC4_KEY *arg1 = (RC4_KEY *) 0 ;
-  PyObject *arg2 = (PyObject *) 0 ;
-  void *argp1 = 0 ;
-  int res1 = 0 ;
-  PyObject * obj0 = 0 ;
-  PyObject * obj1 = 0 ;
-  PyObject *result = 0 ;
-  
-  if(!PyArg_UnpackTuple(args,(char *)"rc4_set_key",2,2,&obj0,&obj1)) SWIG_fail;
-  res1 = SWIG_ConvertPtr(obj0, &argp1,SWIGTYPE_p_RC4_KEY, 0 |  0 );
-  if (!SWIG_IsOK(res1)) {
-    SWIG_exception_fail(SWIG_ArgError(res1), "in method '" "rc4_set_key" "', argument " "1"" of type '" "RC4_KEY *""'"); 
-  }
-  arg1 = (RC4_KEY *)(argp1);
-  {
-    arg2=obj1;
-  }
-  {
-    if (!arg1) {
-      SWIG_exception(SWIG_ValueError,"Received a NULL pointer.");
-    }
-  }
-  result = (PyObject *)rc4_set_key(arg1,arg2);
-  {
-    resultobj=result;
-  }
-  return resultobj;
-fail:
-  return NULL;
-}
-
-
-SWIGINTERN PyObject *_wrap_rc4_update(PyObject *self, PyObject *args) {
-  PyObject *resultobj = 0;
-  RC4_KEY *arg1 = (RC4_KEY *) 0 ;
-  PyObject *arg2 = (PyObject *) 0 ;
-  void *argp1 = 0 ;
-  int res1 = 0 ;
-  PyObject * obj0 = 0 ;
-  PyObject * obj1 = 0 ;
-  PyObject *result = 0 ;
-  
-  if(!PyArg_UnpackTuple(args,(char *)"rc4_update",2,2,&obj0,&obj1)) SWIG_fail;
-  res1 = SWIG_ConvertPtr(obj0, &argp1,SWIGTYPE_p_RC4_KEY, 0 |  0 );
-  if (!SWIG_IsOK(res1)) {
-    SWIG_exception_fail(SWIG_ArgError(res1), "in method '" "rc4_update" "', argument " "1"" of type '" "RC4_KEY *""'"); 
-  }
-  arg1 = (RC4_KEY *)(argp1);
-  {
-    arg2=obj1;
-  }
-  {
-    if (!arg1) {
-      SWIG_exception(SWIG_ValueError,"Received a NULL pointer.");
-    }
-  }
-  result = (PyObject *)rc4_update(arg1,arg2);
-  {
-    resultobj=result;
-  }
-  return resultobj;
-fail:
-  return NULL;
-}
-
-
-SWIGINTERN PyObject *_wrap_rc4_type_check(PyObject *self, PyObject *args) {
-  PyObject *resultobj = 0;
-  RC4_KEY *arg1 = (RC4_KEY *) 0 ;
-  void *argp1 = 0 ;
-  int res1 = 0 ;
-  PyObject * obj0 = 0 ;
-  int result;
-  
-  if(!PyArg_UnpackTuple(args,(char *)"rc4_type_check",1,1,&obj0)) SWIG_fail;
-  res1 = SWIG_ConvertPtr(obj0, &argp1,SWIGTYPE_p_RC4_KEY, 0 |  0 );
-  if (!SWIG_IsOK(res1)) {
-    SWIG_exception_fail(SWIG_ArgError(res1), "in method '" "rc4_type_check" "', argument " "1"" of type '" "RC4_KEY *""'"); 
-  }
-  arg1 = (RC4_KEY *)(argp1);
-  {
-    if (!arg1) {
-      SWIG_exception(SWIG_ValueError,"Received a NULL pointer.");
-    }
-  }
-  result = (int)rc4_type_check(arg1);
-  {
-    resultobj=PyLong_FromLong(result);
-    if (PyErr_Occurred()) SWIG_fail;
-  }
-  return resultobj;
-fail:
-  return NULL;
-}
-
 
 SWIGINTERN PyObject *_wrap_dh_new(PyObject *self, PyObject *args) {
   PyObject *resultobj = 0;
@@ -29635,7 +29489,6 @@
 	 { (char *)"cast5_cbc", _wrap_cast5_cbc, METH_VARARGS, NULL},
 	 { (char *)"cast5_cfb", _wrap_cast5_cfb, METH_VARARGS, NULL},
 	 { (char *)"cast5_ofb", _wrap_cast5_ofb, METH_VARARGS, NULL},
-	 { (char *)"rc4", _wrap_rc4, METH_VARARGS, NULL},
 	 { (char *)"rc2_40_cbc", _wrap_rc2_40_cbc, METH_VARARGS, NULL},
 	 { (char *)"aes_128_ecb", _wrap_aes_128_ecb, METH_VARARGS, NULL},
 	 { (char *)"aes_128_cbc", _wrap_aes_128_cbc, METH_VARARGS, NULL},
@@ -29697,11 +29550,6 @@
 	 { (char *)"AES_set_key", _wrap_AES_set_key, METH_VARARGS, NULL},
 	 { (char *)"AES_crypt", _wrap_AES_crypt, METH_VARARGS, NULL},
 	 { (char *)"AES_type_check", _wrap_AES_type_check, METH_VARARGS, NULL},
-	 { (char *)"rc4_new", _wrap_rc4_new, METH_VARARGS, NULL},
-	 { (char *)"rc4_free", _wrap_rc4_free, METH_VARARGS, NULL},
-	 { (char *)"rc4_set_key", _wrap_rc4_set_key, METH_VARARGS, NULL},
-	 { (char *)"rc4_update", _wrap_rc4_update, METH_VARARGS, NULL},
-	 { (char *)"rc4_type_check", _wrap_rc4_type_check, METH_VARARGS, NULL},
 	 { (char *)"dh_new", _wrap_dh_new, METH_VARARGS, NULL},
 	 { (char *)"dh_free", _wrap_dh_free, METH_VARARGS, NULL},
 	 { (char *)"dh_size", _wrap_dh_size, METH_VARARGS, NULL},
@@ -31111,7 +30959,6 @@
 static swig_type_info _swigt__p_HMAC_CTX = {"_p_HMAC_CTX", "HMAC_CTX *", 0, 0, (void*)0, 0};
 static swig_type_info _swigt__p_PKCS7 = {"_p_PKCS7", "PKCS7 *", 0, 0, (void*)0, 0};
 static swig_type_info _swigt__p_PyObject = {"_p_PyObject", "PyObject *", 0, 0, (void*)0, 0};
-static swig_type_info _swigt__p_RC4_KEY = {"_p_RC4_KEY", "RC4_KEY *", 0, 0, (void*)0, 0};
 static swig_type_info _swigt__p_RSA = {"_p_RSA", "RSA *", 0, 0, (void*)0, 0};
 static swig_type_info _swigt__p_SSL = {"_p_SSL", "SSL *", 0, 0, (void*)0, 0};
 static swig_type_info _swigt__p_SSL_CIPHER = {"_p_SSL_CIPHER", "SSL_CIPHER *", 0, 0, (void*)0, 0};
@@ -31173,7 +31020,6 @@
   &_swigt__p_HMAC_CTX,
   &_swigt__p_PKCS7,
   &_swigt__p_PyObject,
-  &_swigt__p_RC4_KEY,
   &_swigt__p_RSA,
   &_swigt__p_SSL,
   &_swigt__p_SSL_CIPHER,
@@ -31235,7 +31081,6 @@
 static swig_cast_info _swigc__p_HMAC_CTX[] = {  {&_swigt__p_HMAC_CTX, 0, 0, 0},{0, 0, 0, 0}};
 static swig_cast_info _swigc__p_PKCS7[] = {  {&_swigt__p_PKCS7, 0, 0, 0},{0, 0, 0, 0}};
 static swig_cast_info _swigc__p_PyObject[] = {  {&_swigt__p_PyObject, 0, 0, 0},{0, 0, 0, 0}};
-static swig_cast_info _swigc__p_RC4_KEY[] = {  {&_swigt__p_RC4_KEY, 0, 0, 0},{0, 0, 0, 0}};
 static swig_cast_info _swigc__p_RSA[] = {  {&_swigt__p_RSA, 0, 0, 0},{0, 0, 0, 0}};
 static swig_cast_info _swigc__p_SSL[] = {  {&_swigt__p_SSL, 0, 0, 0},{0, 0, 0, 0}};
 static swig_cast_info _swigc__p_SSL_CIPHER[] = {  {&_swigt__p_SSL_CIPHER, 0, 0, 0},{0, 0, 0, 0}};
@@ -31297,7 +31142,6 @@
   _swigc__p_HMAC_CTX,
   _swigc__p_PKCS7,
   _swigc__p_PyObject,
-  _swigc__p_RC4_KEY,
   _swigc__p_RSA,
   _swigc__p_SSL,
   _swigc__p_SSL_CIPHER,
@@ -32133,7 +31977,7 @@
   SWIG_Python_SetConstant(d, d == md ? public_interface : NULL, "PKCS5_SALT_LEN",SWIG_From_int((int)(8)));
   SWIG_addvarlink(SWIG_globals(),(char*)"_evp_err",Swig_var__evp_err_get, Swig_var__evp_err_set);
   SWIG_Python_SetConstant(d, d == md ? public_interface : NULL, "AES_BLOCK_SIZE",SWIG_From_int((int)(AES_BLOCK_SIZE)));
-  SWIG_Python_SetConstant(d, d == md ? public_interface : NULL, "OPENSSL_NO_RC4",SWIG_From_int((int)(0)));
+  SWIG_Python_SetConstant(d, d == md ? public_interface : NULL, "OPENSSL_NO_RC4",SWIG_From_int((int)(1)));
   SWIG_Python_SetConstant(d, d == md ? public_interface : NULL, "dh_check_ok",SWIG_From_int((int)(0)));
   SWIG_Python_SetConstant(d, d == md ? public_interface : NULL, "dh_check_p_not_prime",SWIG_From_int((int)(DH_CHECK_P_NOT_PRIME)));
   SWIG_Python_SetConstant(d, d == md ? public_interface : NULL, "dh_check_p_not_strong",SWIG_From_int((int)(DH_CHECK_P_NOT_STRONG_PRIME)));
